x-pack/plugin/security/qa/rcs-extension/src/main/java/org/elasticsearch/xpack/security/rcs/extension/TestRemoteClusterSecurityExtension.java

```diff
@@ -0,0 +1,125 @@
+import org.elasticsearch.xpack.security.transport.ServerTransportFilter;
+import org.elasticsearch.xpack.security.transport.extension.RemoteClusterSecurityExtension;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+public class TestRemoteClusterSecurityExtension implements RemoteClusterSecurityExtension {
+
+    public static final Setting<String> DUMMY_EXTENSION_SETTING = Setting.simpleString(
+        "xpack.test.rcs.extension.setting",
+        "default-rcs-extension-setting-value",
+        Setting.Property.NodeScope
+    );
+
+    private final Components components;
+
+    public TestRemoteClusterSecurityExtension(Components components) {
+        this.components = components;
+    }
+
+    @Override
+    public RemoteClusterTransportInterceptor getTransportInterceptor() {
+        return new RemoteClusterTransportInterceptor() {
+
+            @Override
+            public TransportInterceptor.AsyncSender interceptSender(TransportInterceptor.AsyncSender sender) {
+                return sender;
+            }
+
+            @Override
+            public boolean isRemoteClusterConnection(Transport.Connection connection) {
+                return false;
+            }
+
+            public boolean hasRemoteClusterAccessHeadersInContext(SecurityContext securityContext) {
+                return false;
+            }
+
+            @Override
+            public Map<String, ServerTransportFilter> getProfileTransportFilters(
+                Map<String, SslProfile> profileConfigurations,
+                DestructiveOperations destructiveOperations
+            ) {
+                Map<String, ServerTransportFilter> profileFilters = Maps.newMapWithExpectedSize(profileConfigurations.size() + 1);
+                Settings settings = components.settings();
+                final boolean transportSSLEnabled = XPackSettings.TRANSPORT_SSL_ENABLED.get(settings);
+
+                for (Map.Entry<String, SslProfile> entry : profileConfigurations.entrySet()) {
+                    final String profileName = entry.getKey();
+                    final SslProfile sslProfile = entry.getValue();
+                    final SslConfiguration profileConfiguration = sslProfile.configuration();
+                    profileFilters.put(
+                        profileName,
+                        new ServerTransportFilter(
+                            components.authenticationService(),
+                            components.authorizationService(),
+                            components.threadPool().getThreadContext(),
+                            transportSSLEnabled && SSLService.isSSLClientAuthEnabled(profileConfiguration),
+                            destructiveOperations,
+                            components.securityContext()
+                        )
+                    );
+                }
+                return Collections.unmodifiableMap(profileFilters);
```

Comment:
Can you add a comment on why this implementation is here?
We don't seem to test this code, so it seems redundant.